libbeat/api/server.go

```
@@ -41,10 +41,6 @@ type Server struct {
 
 // New creates a new API Server with no routes attached.
 func New(log *logp.Logger, config *config.C) (*Server, error) {
-	if log == nil {
-		log = logp.NewLogger("")
-	}
-
```

I think we should return an error if the logger is `nil` but completely removing this validation might be dangerous unless we're 100% sure nothing relies on this logic.

---

So I checked all references using the `New` api. It is safe to remove this